### PR TITLE
Mocked FileWriter: set noaction=1 by default

### LINKED ISSUE #139: Mocked CAF::FileWriter attempts to create directories
I tried to remove `CAF::Object::NoAction=1` in a few unit tests, like `write_grub2_config.t` in AII `aii-pxelinux`. This results in the following exception:
```
Uncaught exception!!! Calling stack is:
  LC::Exception::throw_error called at /usr/lib/perl/LC/Fatal.pm line 261
  LC::Fatal::mkdir called at /usr/lib/perl/LC/File.pm line 529
  LC::File::makedir called at /usr/lib/perl/LC/File.pm line 524
  LC::File::makedir called at /usr/lib/perl/LC/Check.pm line 231
  LC::Check::directory called at /usr/lib/perl/LC/Check.pm line 262
  LC::Check::parent_directory called at /usr/lib/perl/LC/Check.pm line 913
  LC::Check::file called at /usr/lib/perl/CAF/FileWriter.pm line 237
  CAF::FileWriter::close called at /exp/si/jouvin/GitRepositories/Quattor/aii/aii-pxelinux/target/dependency/build-scripts/Test/Quattor.pm line 385
  Test::Quattor::new_filewriter_close called at /exp/si/jouvin/GitRepositories/Quattor/aii/aii-pxelinux/target/lib/perl/NCM/Component/pxelinux.pm line 520
  NCM::Component::pxelinux::_write_grub2_config called at src/test/perl/write_grub2_config.t line 33
  main::check_config called at src/test/perl/write_grub2_config.t line 63
*** mkdir(/grub, 0755): Permission denied
# Tests were run but no plan was declared and done_testing() was not seen.
Uncaught exception!!! Calling stack is:
  LC::Exception::throw_error called at /usr/lib/perl/LC/Fatal.pm line 261
  LC::Fatal::mkdir called at /usr/lib/perl/LC/File.pm line 529
  LC::File::makedir called at /usr/lib/perl/LC/File.pm line 524
  LC::File::makedir called at /usr/lib/perl/LC/Check.pm line 231
  LC::Check::directory called at /usr/lib/perl/LC/Check.pm line 262
  LC::Check::parent_directory called at /usr/lib/perl/LC/Check.pm line 913
  LC::Check::file called at /usr/lib/perl/CAF/FileWriter.pm line 237
  CAF::FileWriter::close called at /exp/si/jouvin/GitRepositories/Quattor/aii/aii-pxelinux/target/dependency/build-scripts/Test/Quattor.pm line 385
  Test::Quattor::new_filewriter_close called at /exp/si/jouvin/GitRepositories/Quattor/aii/aii-pxelinux/target/lib/perl/NCM/Component/pxelinux.pm line 520
  NCM::Component::pxelinux::_write_grub2_config called at src/test/perl/write_grub2_config.t line 33
  main::check_config called at src/test/perl/write_grub2_config.t line 67
*** mkdir(/grub, 0755): Permission denied
src/test/perl/write_grub2_config.t .. 
```
This seems unexpected as `CAF::FileWriter` is supposed to be mocked...

### PR DESCRIPTION
**Pull request by jouvin**
Also:
- Remove `make_directory()` from simple_caf module as it is part of Test/Quattor.pm with a different interface
- Update `directory()` from `Test/Quattor.pm` to return a dualvar
  as the standard `CAF::Path::directory()`
- Update unit tests accordingly

Fixes #139.

### CODE REVIEW

**Comment by jouvin** [comment]
Has been verified to properly work with CAF.

File: build-scripts/src/test/perl/simple_caf.pm [hunk 1]
```diff
@@ -35,18 +35,5 @@ sub make_file
     }
 }
 
-# make a directory
```
Comment (stdweird) [inline build-scripts/src/test/perl/simple_caf.pm]: why is this removed?

Comment (jouvin) [inline build-scripts/src/test/perl/simple_caf.pm]: This is the whole purpose of the PR! This function, if I understand properly, is completely useless as ther is a `make_directory` in `Test/Quattor.pm`. See https://github.com/quattor/maven-tools/pull/150#issue-215843093.

Comment (stdweird) [inline build-scripts/src/test/perl/simple_caf.pm]: the `make_directory` function in `Test::Quattor` is used to make mocked directory.
this module is a test module based on `CAF::Path` (and `CAF::Object`) to be able to test mocked `CAF::Path`.
this is eg how one could/would use it in a component (see the parent relation).

Comment (jouvin) [inline build-scripts/src/test/perl/simple_caf.pm]: I understand the role of `make_directory` in `Test::Quattor` (and it is still there) but I don't understand the role of `make_directory` in `simple_caf`. And in fact removing it doesn't break anything...

**Comment by jouvin** [comment]
Test is failing because I missed a problem when I ran the test locally as I tend not to run the mvn clean: the compilation of the profile (with `use Test::Quattor qw(quattor)`) seems to use the mocked FileWriter/Editor that has now `noaction` set by default instead of using the unmocked FileWriter... I have not been able to figure out where this is done. @stdweird any hint?

File: build-scripts/src/test/perl/simple_caf.pm (hunk 1 above)
Comment (stdweird) [inline build-scripts/src/test/perl/simple_caf.pm]: but `Test::Quattor::make_directory` is something different from `simple_caf->make_directory`, which is an example usecase of `CAF::Path->directory` (which is mocked in `Test::Quattor` using `make_directory`).
wrt "doesn't change anything", well, you also test other/different functions instead. not sure what you expect in such case.

**Comment by stdweird** [comment]
@jouvin i'm trying to make the unittests of the ncm-network cleanup adhere to proper `NoAction` usage (by setting `$Test::Quattor::NoAction=1`), but i'm running in a lot of small mostly unrelated problems.
this needs more testing first.

**Comment by jouvin** [comment]
@stdweird sure it needs to be tested with the various pieces. I think that if we manage to do it, at the end the test framework will be cleaner as there will really be no need to have `CAF::Object::NoAction=1` defined in so many places...

**Comment by jouvin** [comment]
Problem mentioned in https://github.com/quattor/maven-tools/pull/150#issuecomment-290168019 fixed. Should be ready for testing now...

**Comment by stdweird** [comment]
@jouvin i'm adapting and cleaning up the test framework for this case. it's a lot more complicated than this patch, but the new FileWriter allows for cleaner/fewer code, so i'd prefer not to merge this yet. i'll also try to clarify the `make_directory` issue in the PR.

**Comment by jouvin** [comment]
No problem, I always had in mind that this PR required careful evaluation and testing...

**Comment by jrha** [comment]
@jouvin if you have time could you rebase and resolve the conflicts?

**Comment by stdweird** [comment]
@jrha @jouvin unless i'm mistake, this PR was replaced by #160

**Comment by jrha** [comment]
It does indeed seem that way.
